dictionarybar.cc

```diff
@@ -180,7 +180,15 @@ void DictionaryBar::showContextMenu( QContextMenuEvent * event, bool extended )
   if( result && result == headwordsAction )
   {
     QString id = dictAction->data().toString();
-    emit showDictionaryHeadwords( id );
+    // TODO: use `Dictionary::class*` instead of `QString id` as  action->setData -> remove all loops in this file
+    for(auto & dict : allDictionaries)
+    {
+      if( id.compare( dict->getId().c_str() ) == 0 )
+      {
+        emit showDictionaryHeadwords( dict.get() );
+        break;
```

Comment:
id is QString ,compare with a c_str()  ?

Reply:
Fixed with explicit conversion